documentation/v&v/V&V.tex

```diff
@@ -0,0 +1,588 @@
+M5T1 & HTTP POST request to /api/auth/magic containing new MacID & HTTP OK response & Same as expected & Pass \\
+\hline
+M5T2 & HTTP POST request to /api/auth/magic containing existing MacID & HTTP OK response & Same as expected & Pass \\
+\hline
+M5T3 & HTTP POST request to /api/auth/logout & HTTP OK response & Same as expected & Pass \\
+\hline
+\end{tabular}
+\end{table}
+
+\subsection{UserHttpHandler Module (M6)}
+\begin{table}[H]
+\flushleft
+\begin{tabular}{|c|p{4.5cm}|p{3.6cm}|p{3.6cm}|c|}
+\hline
+ \rowcolor{lightgray} 
+\textbf{Test ID} &\textbf{Inputs} &\textbf{Expected Output} &\textbf{Actual Output} &\textbf{Result}\\
+\hline
+M6T1 & Authenticated HTTP GET request to /api/user & HTTP OK response containing current user object & Same as expected & Pass \\
+\hline
+M6T2 & Unauthenticated HTTP GET request to /api/user & HTTP NO\_CONTENT response &Same as expected& Pass \\
+\hline
+M6T3 & Authenticated HTTP GET request to /api/user/:id & HTTP OK response containing matching user object & Same as expected & Pass \\
+\hline
+M6T4 & Unauthenticated HTTP GET request to /api/user/:id & HTTP UNAUTHORIZED response & Same as expected & Pass \\
+\hline
+M6T5 & Authenticated HTTP PUT request to /api/user/:currentUserId containing user update & HTTP OK response containing updated user & Same as expected & Pass \\
+\hline
+M6T6 & Unauthenticated HTTP PUT request to /api/user/:id containing user update & HTTP UNAUTHORIZED response & Same as expected & Pass \\
+\hline
+M6T7 & Authenticated HTTP PUT request to /api/user/:id containing other user update & HTTP FORBIDDEN response & Same as expected & Pass \\
+\hline
+M6T8 & Authenticated HTTP DELETE request to /api/user/:currentUserId & HTTP OK response & Same as expected & Pass \\
+\hline
+M6T9 & Authenticated HTTP DELETE request to /api/user/:id & HTTP FORBIDDEN response & Same as expected & Pass \\
+\hline
+M6T10 & Unauthenticated HTTP DELETE request to /api/user/:id & HTTP UNAUTHORIZED response & Same as expected & Pass \\
+\hline
+\end{tabular}
+\end{table}
+
+\section{More Than the Tests}
+\subsection{Code Reviews}
+
+All code pushed to the Git repository was peer-reviewed by other members of the team. New code had to be reviewed and approved by at least one other member before merging into the main branch.\\
+History of all peer-reviewed merge results can be found \href{https://github.com/calebmech/Book-Bazar/pulls?q=is\%3Apr+is\%3Amerged}{on GitHub}.
+
+\subsection{Static Analysis}
+Static analysis was performed using eslint. Common JavaScript and React pitfalls were avoided using eslint and code styling was enforced by the Prettier code formatting tool. We run eslint as a part of our continuous integration testing to ensure that new code being added to the main repository adheres to the code style. Furthermore, type safety at the static analysis level was added by using TypeScript over vanilla JavaScript.
+
+\section{Supporting Material}
+Please refer to prior design documents submitted.
```

Comment:
I think we should modify this wording to add the motivation for "Why should I look at the previous documentation?" "What will I find in the previous documentation that isn't here?"